### PR TITLE
feat: Disable creation of short top-level account ids

### LINKED ISSUE #2292: Disable Top Level Account registration
We should disable TLA registration unless it's done by account name `registrar`.
The goal is to have an auction later to register top level account names. 
We are not implementing that for the launch though.

I think we can allow to register accounts, which names are longer than 10 (or 32?) characters.

`registrar` account originally will be controlled by Foundation. Foundation will later deploy the naming auction contract on it.

Implementing https://github.com/nearprotocol/NEPs/pull/45

CC @kcole16

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Disabling creation of top-level account IDs shorted than 11 characters, except if created by `registrar`.
- Adding a new error that tells that the account has to be created by `registrar`
- Adding config to do this
- Adding migration script
- Bumping protocol version
- Adding `registrar` account

NEP: https://github.com/nearprotocol/NEPs/pull/45
Fixes #2292

## Test plan:
- CI
- State migration test
- Extra tests for account ID creation.

### CODE REVIEW

File: core/primitives/src/errors.rs [hunk 1]
```diff
@@ -302,6 +302,12 @@ pub enum ActionErrorKind {
     AccountAlreadyExists { account_id: AccountId },
     /// Happens when TX receiver_id doesn't exist (but action is not Action::CreateAccount)
     AccountDoesNotExist { account_id: AccountId },
+    /// A newly created top-level account ID can only be created by registrar.
```
Comment (nearmax) [inline core/primitives/src/errors.rs:305]:
```suggestion
    /// A top-level account ID can only be created by registrar.
```

File: core/primitives/src/errors.rs [hunk 2]
```diff
@@ -597,10 +602,15 @@ impl Display for ActionErrorKind {
...
+            ActionErrorKind::CreateAccountOnlyByRegistrar { account_id, registrar_account_id, predecessor_id } => write!(
+                f,
+                "A new top-level account ID {:?} can't be created by {:?}, short top-level account IDs can only be created by {:?}",
+                account_id, predecessor_id, registrar_account_id,
+            ),
             ActionErrorKind::CreateAccountNotAllowed { account_id, predecessor_id } => write!(
                 f,
-                "The new account_id {:?} can't be created by {:?}",
-                account_id, predecessor_id
+                "A new sub-account ID {:?} can't be created by account {:?}",
```
Comment (nearmax) [inline core/primitives/src/errors.rs:612]:
```suggestion
                "A sub-account ID {:?} can't be created by account {:?}",
```

File: core/primitives/src/errors.rs [hunk 3]
```diff
@@ -597,10 +602,15 @@ impl Display for ActionErrorKind {
             ActionErrorKind::UnsuitableStakingKey { public_key } => {
                 write!(f, "The staking key must be ED25519. {} is provided instead.", public_key)
             }
+            ActionErrorKind::CreateAccountOnlyByRegistrar { account_id, registrar_account_id, predecessor_id } => write!(
+                f,
+                "A new top-level account ID {:?} can't be created by {:?}, short top-level account IDs can only be created by {:?}",
```
Comment (nearmax) [inline core/primitives/src/errors.rs:607]:
```suggestion
                "A top-level account ID {:?} can't be created by {:?}, short top-level account IDs can only be created by {:?}",
```

**Comment by evgenykuzyakov** [comment]
> How do we resolve its impact on testnet?

Testnet will not be able to create short top-level account IDs. No other issues.

File: core/runtime-configs/src/lib.rs [hunk 4]
```diff
@@ -38,6 +41,29 @@ impl RuntimeConfig {
             storage_amount_per_byte: 0,
             transaction_costs: RuntimeFeesConfig::free(),
             wasm_config: VMConfig::free(),
+            account_creation_config: AccountCreationConfig {
+                min_allowed_top_level_account_length: 0,
```
Comment (nearmax) [inline core/runtime-configs/src/lib.rs:45]: This shouldn't be different than default.

File: scripts/migrations/7-account-registrar.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        near_access_key_records.append(record['AccessKey'])
+    records.append(record)
+
+assert(len(near_access_key_records) > 0)
+
+records.append({
+    'Account': {
+        'account': {
+            # 1_000_000 N or 1e30
+            'amount': '1' + ('0' * 30),
```
Comment (frol) [inline scripts/migrations/7-account-registrar.py:51]: `str(10**30)` looks more readable to me.

File: pytest/lib/branches.py [hunk 6]
```diff
@@ -54,7 +54,7 @@ def download_binary(branch):
 
 def prepare_ab_test(other_branch):
     compile_current()
-    if other_branch in ['master', 'beta', 'stable']:
+    if os.environ.get('BUILDKITE') and other_branch in ['master', 'beta', 'stable']:
```
Comment (frol) [inline pytest/lib/branches.py:57]: So, does this mean that downloading a prebuild binary makes sense only in the BuildKite environment? /cc @ailisp

Comment (evgenykuzyakov) [reply pytest/lib/branches.py:57]: Yeah, on mac they don't quite work :)

Comment (ailisp) [reply pytest/lib/branches.py:57]: It also makes sense for local linux, and soon mac (#2299 ) I'll change this after #2299 being build for both master and beta.

**Comment by evgenykuzyakov** [comment]
@bowenwang1996 @SkidanovAlex Is there any `nightly` tests that require creation short account IDs?

**Comment by bowenwang1996** [comment]
> Testnet will not be able to create new short top-level account IDs. No other issues.

How is this not an issue? It seems to me that this will make user experience worse, both for end users and app developers who may want to create account not under their account name. I think we shouldn't impose unnecessary restrictions on testnet, just as we won't disallow token transfers.

**Comment by evgenykuzyakov** [comment]
Instead of reverting this PR completely, we can shorted the length allowance in the config: #2362

**Pull request merged**